ocaml/xapi/helpers.ml

```diff
@@ -250,6 +250,26 @@ let make_rpc ~__context rpc : Rpc.response =
     else SSL(SSL.make ~use_stunnel_cache:true (), Pool_role.get_master_address(), !Xapi_globs.https_port) in
   XMLRPC_protocol.rpc ~srcstr:"xapi" ~dststr:"xapi" ~transport ~http rpc
 
+let make_timeboxed_rpc ~__context timeout rpc : Rpc.response =
+  let subtask_of = Ref.string_of (Context.get_task_id __context) in
+  Server_helpers.exec_with_new_task "timeboxed_rpc" ~subtask_of:(Context.get_task_id __context) (fun __context ->
```

Comment:
We call this from `events_from_xapi`, which calls `Server_helpers.exec_with_new_task "xapi events"` followed by `Helpers.call_api_functions`, and now again `Server_helpers.exec_with_new_task`. This is at least confusing, perhaps worse...?

`make_rpc` ⬆️ does not seem to need it?

Reply:
Right, that's because I need to associate the 'resources' with the temporary task so I can kill the stunnel independently of any other resources associated with the task.

Reply:
Aha, interesting. It would be nice to have some comments in the code about this (also about the partial application).

Reply:
Comments added.